dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -678,8 +678,11 @@ public ResourceSupport findRel(HttpServletRequest request, HttpServletResponse r
                     if (Page.class.isAssignableFrom(linkMethod.getReturnType())) {
                         Page<? extends RestModel> pageResult = (Page<? extends RestAddressableModel>) linkMethod
                                 .invoke(linkRepository, request, uuid, page, projection);
-                        Link link = linkTo(this.getClass(), apiCategory, model).slash(uuid).slash(subpath)
-                                .withSelfRel();
+
+                        //TODO we need to strip any existing paging information from the query string
+                        //Even if the paging info is not removed, the params don't get duplicated.
+                        Link link = linkTo(this.getClass(), apiCategory, model).slash(uuid)
+                                .slash(subpath + '?' + request.getQueryString()).withSelfRel();
```

Comment:
if the query string is null or empty the question mark should be not added

Reply:
Added new test -- startsWithandPage